pkg/image/registry/imagestreamimport/rest.go

```diff
@@ -199,6 +200,18 @@ func (r *REST) Create(ctx apirequest.Context, obj runtime.Object, createValidati
 		return nil, kapierrors.NewInternalError(err)
 	}
 
+	// check imported images status. If we get authentication error (401), try import same image without authentication.
+	// Docker registry gives 401 on public images if you have wrong secret in your secret list.
+	for _, image := range isi.Status.Images {
+		if image.Status.Reason == metav1.StatusReasonUnauthorized && strings.Contains(image.Status.Message, "incorrect username or password") {
+			importCtx := registryclient.NewContext(r.transport, r.insecureTransport).WithCredentials(nil)
+			imports := r.importFn(importCtx)
+			if err := imports.Import(ctx.(gocontext.Context), isi, stream); err != nil {
+				return nil, kapierrors.NewInternalError(fmt.Errorf(image.Status.Message))
```

Comment:
on further analysis i don't think this is correct.  "err" will still be nil, even if the import failed due to credentials, no?

just as the credentials import check above expects err to be nil even if the import was disallowed by auth.

So you're turning a more serious error that occurred during import, into a credential message.  I don't think that's right.

Reply:
I think I overthinking this... I think let's leave previous error. Standart error when image credentials are wrong is for private image is:
```
! error: Import failed (Unauthorized): you may not have access to the Docker image "docker.io/juozasa/hellonode:latest"                                                                                          
      Less than a second ago      
```
Only when public image is being imported it gives the wrong password when trying with auth header. So only in this case (as much as I manage to test all possible cases), it defaults to this condition. Its just how docker distribution api behaves...

Reply:
or we can concatenate both errors to one string but I would better keep it simple... previous way.

Reply:
i agree the error should be returned as is.

i'm less clear on what image.status.message is going to end up being in this case (where you provided invalid credentials and the image is not public, but we tried to access it with nil credentials)?